demo/api/src/warnings/warning-checker.console.ts

```diff
@@ -0,0 +1,78 @@
+import { BlockInput, FlatBlocks } from "@comet/blocks-api";
+import { DiscoverService } from "@comet/cms-api/lib/dependencies/discover.service";
+import { CreateRequestContext, MikroORM } from "@mikro-orm/core";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { EntityManager, EntityRepository } from "@mikro-orm/postgresql";
+import { Injectable } from "@nestjs/common";
+import { Command, Console } from "nestjs-console";
+import { v5 } from "uuid";
+
+import { Warning } from "./entities/warning.entity";
+
+@Injectable()
+@Console()
+export class WarningCheckerConsole {
+    constructor(
+        private readonly orm: MikroORM,
+        private readonly discoverService: DiscoverService,
+        private readonly entityManager: EntityManager,
+        @InjectRepository(Warning) private readonly warningsRepository: EntityRepository<Warning>,
+    ) {}
+
+    @Command({
+        command: "check-warnings",
+        description: "Checks for warnings",
+    })
+    @CreateRequestContext()
+    async execute(): Promise<void> {
+        // TODO: (in the next PRs) Check if data itself is valid in the database. (Maybe some data was put into database and is not correct or a migration was done wrong)
+
+        for (const rootBlockEntity of this.discoverService.discoverRootBlocks()) {
+            const { metadata, column, block } = rootBlockEntity;
+
+            const queryBuilder = this.entityManager.createQueryBuilder(metadata.className);
+            queryBuilder.select(`id, "${column}"`).from(metadata.tableName);
+            const rootBlocks = (await queryBuilder.getResult()) as Array<{ [key: string]: BlockInput }>;
+
+            for (const rootBlock of rootBlocks) {
+                const blockData = block.blockDataFactory(block.blockInputFactory(rootBlock[column]));
+
+                const flatBlocks = new FlatBlocks(blockData, {
+                    name: block.name,
+                    visible: true,
+                    rootPath: "root",
+                });
+                for (const node of flatBlocks.depthFirst()) {
+                    const warnings = node.block.warnings();
+
+                    if (warnings.length > 0) {
+                        // TODO: (in the next PRs) auto resolve warnings
+
+                        for (const warning of warnings) {
+                            const type = "Block";
+                            const staticNamespace = "4e099212-0341-4bc8-8f4a-1f31c7a639ae";
+                            const id = v5(`${metadata.tableName}${rootBlock["id"]};${warning.message}`, staticNamespace);
```

Comment:
clever concept. however a message might not be the best to use here

Reply:
The Problem is I somehow have to distinguish different warnings in the same block. Can you think of another way to distinguish the warning other than via the message?